instance/models/instance.py

```diff
@@ -586,4 +697,41 @@ def provision(self):
 
         return (server, ansible_log)
 
+    def provision_external_databases(self):
+        """
+        Provision external databases for this instance
+        """
+        self.provision_external_mysql()
+        self.provision_external_mongo()
+
+    def provision_external_mysql(self):
+        """
+        Provision external mysql database for this instance
+        """
+        if settings.INSTANCE_MYSQL_URL_OBJ:
+            connection = mysql.connect(
+                host=settings.INSTANCE_MYSQL_URL_OBJ.hostname,
+                user=settings.INSTANCE_MYSQL_URL_OBJ.username,
+                passwd=settings.INSTANCE_MYSQL_URL_OBJ.password or '',
+                port=settings.INSTANCE_MYSQL_URL_OBJ.port or 3306,
+            )
+            cursor = connection.cursor()
+
+            # We can't use the database name in a parameterized query, the
+            # driver doesn't escape it properly. Se we escape it here instead
+            database_name = connection.escape_string(self.database_name).decode()
+
+            cursor.execute('CREATE DATABASE IF NOT EXISTS {0}'.format(database_name))
+            cursor.execute('GRANT ALL ON {0}.* TO %s IDENTIFIED BY %s'.format(database_name),
+                           (self.database_user, self.database_pass))
+
+    def provision_external_mongo(self):
+        """
+        Provision external mongo databases for this instance
+        """
+        if settings.INSTANCE_MONGO_URL:
+            mongo = pymongo.MongoClient(settings.INSTANCE_MONGO_URL)
+            for database in (self.database_name, self.forum_database_name):
```

Comment:
@omarkhan Ah, I see, you actually reuse the same DB name for both MySQL and mongo - just using a suffix for the 2nd DB for a given driver. That makes sense, actually - I would be fine keeping it, if you find a way to address my comment about sanitizing the db name in the `self.database_name` property to not need escaping, in a way that works across DB.